pkg/migrations/op_add_column.go

```diff
@@ -52,18 +53,16 @@ func (o *OpAddColumn) Start(ctx context.Context, conn *sql.DB, stateSchema strin
 			SQL:            *o.Up,
 		})
 		if err != nil {
-			return fmt.Errorf("failed to create trigger: %w", err)
-		}
-		if err := backfill(ctx, conn, table, cbs...); err != nil {
-			return fmt.Errorf("failed to backfill column: %w", err)
+			return nil, fmt.Errorf("failed to create trigger: %w", err)
 		}
+		tableToBackfill = table
 	}
 
 	table.AddColumn(o.Column.Name, schema.Column{
```

Comment:
wondering if this could cause issues, we return a pointer to the table to backfill, then we add the new column to it. We are effectively adding a column here, and it could be a NOT NULL UNIQUE column, so backfill could potentially decide to use this one to backfill the very same column 😇 , which would create issues. Wondering how this can be solved, probably backfill can be smart enough to skip the backfilled column when choosing the identity

Reply:
It's a good point about adding a `NOT NULL`, `UNIQUE` column here and then having `Backfill` decide to use it as the identity column for backfilling.

I don't think it's going to be a big issue in practice as `Backfill` will prefer to use a PK if one exists on the table.

I think it should be possible to have `Backfill` decline to use columns that have a `__pgroll_new` name as the identity. I can do that in a follow up PR if that works for you.

Reply:
approved but please don't forget about this, I believe it's possible to get into an infinite loop with the current code